src/formatter/text-format.tsx

```diff
@@ -32,6 +32,11 @@ export const TextFormat = ({ value, type, format, blankOnInvalid, locale }: ITex
   if (!locale) {
     // TODO: find a better way to keep track of *current* locale
     locale = TranslatorContext.context.locale;
+
+    if (!numeral.locales.includes(locale)) {
+     // if not include, by default as en
+     locale = 'en';
+    }
```

Comment:
This might affect date format as well since the variable is in outer scope. May be better to directly set numerla locale here and set actual locale in else

Reply:
I think at least the program is working properly to consider these, or here can give a warning lack of relevant locale, not nothing to do. The whole page is wrong because the internal dependency is missing locale, which is confusing for users trying jhipster for the first time

Reply:
What I meant is that solution is fine but it shouldn't affect dates so do this

```
if (!numeral.locales.includes(locale)) {
     // if not include, by default as en
     numeral.locale('en'); 
} else {
numeral.locale(locale);
}
```

Reply:
Is numeral locales an array? Tests are failing with a type error

Reply:
it is map